From ef6665ec410fe89af700e3257942083f4ea629a8 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 27 Jun 2014 18:05:37 +0200
Subject: [PATCH] base/init.sh: use loop counter with unique varname

$i can be polluted too easily

(cherry picked from commit 6525182bfc06abfffd71dafc06c5faca6c55eace)
---
 modules.d/99base/init.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules.d/99base/init.sh b/modules.d/99base/init.sh
index 9ee3e9dc..dcc62d2a 100755
--- a/modules.d/99base/init.sh
+++ b/modules.d/99base/init.sh
@@ -231,7 +231,7 @@ source_hook pre-mount
 getarg 'rd.break=mount' -d 'rdbreak=mount' && emergency_shell -n mount "Break mount"
 # mount scripts actually try to mount the root filesystem, and may
 # be sourced any number of times. As soon as one suceeds, no more are sourced.
-i=0
+_i_mount=0
 while :; do
     if ismounted "$NEWROOT"; then
         usable_root "$NEWROOT" && break;
@@ -247,8 +247,8 @@ while :; do
         fi
     done
 
-    i=$(($i+1))
-    [ $i -gt 20 ] \
+    _i_mount=$(($_i_mount+1))
+    [ $_i_mount -gt 20 ] \
         && { flock -s 9 ; action_on_fail "Can't mount root filesystem" && break; } 9>/.console_lock
 done
 
